app/controllers/api/v1/reports_controller.rb

```diff
@@ -0,0 +1,24 @@
+module Api
+  module V1
+    # reports endpoint.  No swagger docs since this is admin-user only
+    # This returns tsv files
+    class ReportsController < ApiBaseController
+      include ActionController::RequestForgeryProtection
+
+      before_action :authenticate_api_user!
+
+      def show
+        if !current_api_user.admin?
+          head 403 and return
+        end
+
+        report_name = params[:report_name].to_sym
+        if !ReportsService::REPORTS.keys.include?(report_name)
```

Comment:
Since `ReportsService.get_report_data` will raise an error if the requested report doesn't exist, I'd leave this out and do this instead:

```
begin
  render plain: ReportsService.get_report_data(report_name)
rescue ArgumentError => e
  render json: {error: e.message}, status: 422
end
```

and change `reports_service.rb` line 16 to:

```
if report_obj.nil?
  raise ArgumentError.new("Unrecognized report: '#{report_name}'")
end
```